src/runtime/vulkan_memory.h

```diff
@@ -0,0 +1,721 @@
+#ifndef HALIDE_RUNTIME_VULKAN_MEMORY_H
+#define HALIDE_RUNTIME_VULKAN_MEMORY_H
+
+#include "internal/block_allocator.h"
+#include "vulkan_internal.h"
+
+namespace Halide {
+namespace Runtime {
+namespace Internal {
+namespace Vulkan {
+
+// --------------------------------------------------------------------------
+
+// Enable external client to override Vulkan allocation callbacks (if they so desire)
+WEAK ScopedSpinLock::AtomicFlag custom_allocation_callbacks_lock = 0;
+static const VkAllocationCallbacks *custom_allocation_callbacks = nullptr;  // nullptr => use Vulkan runtime implementation
```

Comment:
It's generally not a good idea to declare something as `static` in a .h file, since you'll end up with a separate static instance in every .cpp file you include this in -- if that is actually what you *want* here you really need to add comments here as to why that is desirable and intentional (but even then it seems dubious0